bundles/org.openhab.binding.dirigera/src/main/java/org/openhab/binding/dirigera/internal/handler/DirigeraHandler.java

```diff
@@ -0,0 +1,1056 @@
+            locationOptions.add(new CommandOption(location.toFullString(), "Home location"));
+        }
+        commandProvider.setCommandOptions(new ChannelUID(thing.getUID(), CHANNEL_LOCATION), locationOptions);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (customDebug) {
+            logger.info("DIRIGERA HANDLER command {} : {} {}", channelUID, command.toFullString(), command.getClass());
+        }
+        String channel = channelUID.getIdWithoutGroup();
+        if (command instanceof RefreshType) {
+            State cachedState = channelStateMap.get(channel);
+            if (cachedState != null) {
+                super.updateState(channelUID, cachedState);
+            }
+        } else if (CHANNEL_PAIRING.equals(channel)) {
+            JSONObject permissionAttributes = new JSONObject();
+            permissionAttributes.put(PROPERTY_PERMIT_JOIN, OnOffType.ON.equals(command));
+            api().sendAttributes(config.id, permissionAttributes);
+        } else if (CHANNEL_LOCATION.equals(channel)) {
+            PointType coordinatesPoint = null;
+            if (command instanceof PointType point) {
+                coordinatesPoint = point;
+            } else if (command instanceof StringType string) {
+                if (string.toFullString().isBlank()) {
+                    String nullCoordinates = model().getTemplate(Model.TEMPLATE_NULL_COORDINATES);
+                    JSONObject patchCoordinates = new JSONObject(nullCoordinates);
+                    if (customDebug) {
+                        logger.info("DIRIGERA HANDLER send null coordinates {}", patchCoordinates);
+                    }
+                    api().sendPatch(config.id, patchCoordinates);
+                } else {
+                    try {
+                        coordinatesPoint = new PointType(string.toFullString());
+                    } catch (IllegalArgumentException exception) {
+                        logger.info("DIRIGERA HANDLER wrong home location format {} : {}", string,
+                                exception.getMessage());
+                    }
+                }
+            }
+            if (coordinatesPoint != null) {
+                String coordinatesTemplate = model().getTemplate(Model.TEMPLATE_COORDINATES);
+                String coordinates = String.format(coordinatesTemplate, coordinatesPoint.getLatitude().toFullString(),
+                        coordinatesPoint.getLongitude().toFullString());
+                if (customDebug) {
+                    logger.info("DIRIGERA HANDLER send coordinates {}", coordinates);
+                }
+                api().sendPatch(config.id, new JSONObject(coordinates));
+            }
+        }
+    }
+
+    @Override
+    public void initialize() {
+        // do this asynchronous in case of token and other parameters needs to be
+        // obtained via Rest API calls
+        scheduler.execute(this::doInitialize);
```

Comment:
The sequence should be something like:
1. check user settable configuration and set the state accordingly
2. set the state to unknown
3. start the async init task to 
a. get the token (set state if it fails)
b. do whatever needs to be done and scheduled the task to update values.
c. as part of the values being updated set the state to online.

The current implementation is close, so it won;t be that much of work.

Reply:
Did you see this?

Reply:
Yes, reworked